src/tools/genpolicy/rules.rego

```diff
@@ -55,14 +53,178 @@ default AllowRequestsFailingPolicy := false
+    every env_key, env_val in i_env_map {
+      print("allow_env: env_key =", env_key, "env_val =", env_val)
+      allow_env_map_entry(env_key, env_val, p_env_map)
+    }
+    print("allow_env_map: true")
+}
+
+# Allow exact match
+allow_env_map_entry(key, i_val, p_env_map) {
+    p_val := p_env_map[key]
+    i_val == p_val
+    print("allow_env_map_entry: true")
+}
+
+# Allow variables that should look like a subdomain name
+allow_env_map_entry(key, i_val, p_env_map) {
+  p_val := p_env_map[key]
+  some allowed in ALLOWED_SUBDOMAIN_NAMES
+  p_val == allowed
+  regex.match(SUBDOMAIN_NAME, i_val)
+  print("allow_env_map_entry2: true")
+}
 
-CreateContainerRequest:= {"ops": ops, "allowed": true} {
+# Allow input env variables that match with a request_defaults regex.
+allow_env_map_entry(key, i_val, p_env_map) {
+    some p_regex1 in policy_data.request_defaults.CreateContainerRequest.allow_env_regex
+    p_regex2 := replace(p_regex1, "$(ipv4_a)", policy_data.common.ipv4_a)
+    p_regex3 := replace(p_regex2, "$(ip_p)", policy_data.common.ip_p)
+    p_regex4 := replace(p_regex3, "$(svc_name)", policy_data.common.svc_name)
+    p_regex5 := replace(p_regex4, "$(dns_label)", policy_data.common.dns_label)
+
+    result := concat("=", [key, i_val])
+    regex.match(p_regex5, result)
+
+    print("allow_env_map_entry 3: true")
+}
+
+# Allow fieldRef "fieldPath: status.podIP" values.
+allow_env_map_entry(key, i_val, p_env_map) {
+    is_ip(i_val)
+
+    p_val := p_env_map[key]
+    p_var := concat("=", [key, p_val])
+    allow_pod_ip_var(key, p_var)
+    print("allow_env_map_entry 4: true")
+}
+
+# Match input with one of the policy variables, after substituting $(sandbox-name).
+allow_env_map_entry(key, i_val, p_env_map) {
+    s_name := input.base.OCI.Annotations[S_NAME_KEY]
+    p_val := p_env_map[key]
+    p_var2 := replace(p_val, "$(sandbox-name)", s_name)
+    p_var2 == i_val
+    print("allow_env_map_entry 5: true")
+}
+
+# Match input with one of the policy variables, after substituting $(sandbox-namespace).
+allow_env_map_entry(key, i_val, p_env_map) {
+    s_namespace := input.base.OCI.Annotations[S_NAMESPACE_KEY]
+    p_val := p_env_map[key]
+    p_var2 := replace(p_val, "$(sandbox-namespace)", s_namespace)
+    p_var2 == i_val
+    print("allow_env_map_entry 6: true")
+}
+
+# Allow fieldRef "fieldPath: status.hostIP" values.
+allow_env_map_entry(key, i_val, p_env_map) {
+    is_ip(i_val)
+
+    p_val := p_env_map[key]
+    p_var := concat("=", [key, p_val])
+    allow_host_ip_var(key, p_var)
+    print("allow_env_map_entry 7: true")
+}
+
+# Allow resourceFieldRef values (e.g., "limits.cpu").
+allow_env_map_entry(key, i_val, p_env_map) {
+    p_val := p_env_map[key]
+    # TODO: should these be handled in a different way?
+    some allowed in ALWAYS_ALLOWED
+    p_val == allowed
+    #regex.match(SUBDOMAIN_NAME, i_val)
+    print("allow_env_map_entry8: true")
+}
+
+CreateContainerRequest:= resp {
+  not input.env_map
```

Comment:
This adds backport support for azl2 images that may not get a new kata version needed to transform the input. This also removes the need for this version to be released in sync with a new agent version

Reply:
Do we need this? For AzL2 we're merely cutting policy fixes if really need be. As we retire AzL2 soon, we may not want to add this?

Reply:
@ms-mahuber , I believe the idea here is to keep the older rules for a while, that still work when using an older Agent. These are not newly-added rules.